binaryset/binary_set.h

```diff
@@ -0,0 +1,22 @@
+#include <stdio.h>
+#include <stdlib.h>
+ 
+typedef int (*Comp)(const void*, const void*);
+
+struct BinarySetElement
+{
+    void *a;
+    struct BinarySetElement *left;
+    struct BinarySetElement *right;
+};
+
+struct BinarySet{
```

Comment:
Public structs should be defined with `typedef` and name of the struct should be in snake case and include "rtw" as prefix. Eg.
```c
typedef struct {
   // ...
} rtw_bset;
````